src/client/WebSocketClient.test.ts

```diff
@@ -219,6 +219,96 @@ describe('WebSocketClient', () => {
       return ws;
     }
 
+    function setupWebSocketServer(payload?: Object): void {
+      server.on('connection', (ws: WebSocket) => {
+        ws.on('message', (message: string) => {
+          const request = JSON.parse(message) as WebSocketRequest;
+
+          if (request.type === WebSocketRequestType.SUBSCRIBE) {
+            // Send subscription confirmation
+            server.clients.forEach(client =>
+              client.send(
+                JSON.stringify({
+                  channels: request.channels,
+                  type: WebSocketResponseType.SUBSCRIPTIONS,
+                })
+              )
+            );
+
+            if (payload) {
+              // Send event for subscription
+              server.clients.forEach(client => client.send(JSON.stringify(payload)));
+            }
+          }
+
+          if (request.type === WebSocketRequestType.UNSUBSCRIBE) {
+            // Send unsubscribe confirmation
+            server.clients.forEach(client => client.send(JSON.stringify(emptySubscriptions)));
+          }
+        });
+      });
+    }
+
+    function mockWebSocketClientSubscription(
```

Comment:
I wrote this function so that it could replace the `mockWebSocketResponse()` function. However, I did not replace that function in this PR as that's out of the scope of this PR. The `mockWebSocketResponse()` could not work for my needs here because it makes some assumptions that wouldn't allow me to properly test this functionality.